### PR TITLE
Add creator/author on creation of Content Block Edition

### PR DESCRIPTION
**Pull request by Harriethw**
When I create a new Content Block Edition
Then I am added as the first Author of that edition
And I am the Creator of that edition

This follows existing pattern of Editions in the main Whitehall code.

object_store_models

### CODE REVIEW

**Review by pezholio — COMMENTED** [review]
Looking on the right track, ta! Just one failing test and then I think it's ready for a proper review

File: lib/engines/content_object_store/app/models/content_object_store/content_block_edition.rb [hunk 1]
```diff
@@ -1,3 +1,20 @@
 class ContentObjectStore::ContentBlockEdition < ApplicationRecord
   include ContentObjectStore::Identifiable
+
+  has_many :content_block_edition_authors, dependent: :destroy
```
[the remark this answers is not in the document]
Comment (Harriethw) [reply lib/engines/content_object_store/app/models/content_object_store/content_block_edition.rb:4]: I can see how tidying away functionality into Concerns makes the class smaller, but I feel a bit like I'm missing something as I'm finding it hard to adjust to Ruby after a year of typed languages - when you're coding how do you quickly grok the shape and meaning of a Class? E.g. in Kotlin or Typescript I would easily know the fields, behaviour and methods by just looking at the class file or type definition. But with Ruby if we are abstracting behaviour and even the high level fields and relationships to Concerns, are you clicking through every Concern? And looking at the `schema.db` to understand the table data?

[the remark this answers is not in the document]
Comment (pezholio) [reply lib/engines/content_object_store/app/models/content_object_store/content_block_edition.rb:4]: Yeah, basically. It's not ideal, but I do find it easier to deal with crowded models in that way. I guess with Spring Boot, the model is basically describing only the data layer, whereas with Rails, you have behaviour-related stuff in your model, and you have to look at the schema to actually see what the shape of the object is.

File: lib/engines/content_object_store/app/models/concerns/content_object_store/hasAuthors.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 module ContentObjectStore
   module HasAuthors
     extend ActiveSupport::Concern
+    include ContentObjectStore::HasCreator
```
Comment (Harriethw) [inline lib/engines/content_object_store/app/models/concerns/content_object_store/hasAuthors.rb:4]: @pezholio I'm popping the `HasCreator` into `HasAuthors` as it directly depends on the Edition already having authors, but maybe it should be included on the Edition itself?

**Review by pezholio — APPROVED** [review]
Looks great ta! 👍

**Pull request merged**